lib/doc/worksheet.js

```diff
@@ -334,18 +334,57 @@ class Worksheet {
     return row;
   }
 
-  addRow(value) {
-    const row = this.getRow(this._nextRow);
+  addRow(value, styleOption = 'n') {
+    const rowNo = this._nextRow;
+    const row = this.getRow(rowNo);
     row.values = value;
+    this._setStyleOption(rowNo, styleOption === 'i' ? styleOption : 'n');
     return row;
   }
 
-  addRows(value) {
+  addRows(value, styleOption = 'n') {
     value.forEach(row => {
-      this.addRow(row);
+      this.addRow(row, styleOption);
     });
   }
 
+  insertRow(pos, value, styleOption = 'n') {
+    this.spliceRows(pos, 0, value);
+    this._setStyleOption(pos, styleOption);
+    return this.getRow(pos);
+  }
+
+  insertRows(pos, values, styleOption = 'n') {
+    this.spliceRows(pos, 0, ...values);
+    if (styleOption !== 'n') {
```

Comment:
@Siemienik I remember we had a discussion around cloning/copying style objects - could you review this part of the new code?

Reply:
I think that the best idea to maintain style's is change logic of setters to use Object.freez and copy values there. ( When any setter style is being used then all style object should be copy and freez)

I didn't know yet how this solution impact to this pr.

Reply:
Style setting logic is based on already existing logic in splice method. 

`rDst.style = rSrc.style;
    // eslint-disable-next-line no-loop-func
    rSrc.eachCell({includeEmpty: true}, (cell, colNumber) => {
      rDst.getCell(colNumber).style = cell.style;
    });
    rDst.height = rSrc.height;`

Are you suggesting something like, this:
`rDst.style = Object.freeze({...rSrc.style});
    // eslint-disable-next-line no-loop-func
    rSrc.eachCell({includeEmpty: true}, (cell, colNumber) => {
      rDst.getCell(colNumber).style = Object.freeze({...cell.style});
    });
    rDst.height = rSrc.height;`

Reply:
@Siemienik are you saying, to do it like this:
rDst.getCell(colNumber).style = Object.freeze({...cell.style}); });

Reply:
@Siemienik the existing property specific setter actual logic is in "_applyStyle" in row.js and again in cell's setter. This is not using Object.freeze yet. This may be added in another PR.
This pr directly copies style internally, just like spilceRows. 
I have update the copy logic with Object.freeze as you told.
@Siemienik @alubbe Please review. Thank you.